de.dlr.sc.virsat.swtbot.test/src/de/dlr/sc/virsat/swtbot/test/FuncElectricalDiagramTest.java

```diff
@@ -0,0 +1,189 @@
+	}
+	
+	@Test
+	public void removeInterfaceDiagramElementUndoRedoTest() {
+		addElement(Interface.class, conceptFuncElectrical, elementConfiguration);
+		dragTreeItemToDiagramEditor(elementConfiguration, diagramEditor);
+		Assert.assertTrue(isEditPartPresentInDiagramEditor(diagramEditor, "ElementConfiguration"));
+		
+		removeEditPartInDiagramEditor(diagramEditor, "ElementConfiguration");
+		Assert.assertFalse(isEditPartPresentInDiagramEditor(diagramEditor, "ElementConfiguration"));		
+		
+		undoCommand();
+		Assert.assertTrue(isEditPartPresentInDiagramEditor(diagramEditor, "ElementConfiguration"));
+		redoCommand();
+		Assert.assertFalse(isEditPartPresentInDiagramEditor(diagramEditor, "ElementConfiguration"));
+	}
+	
+	@Test
+	public void deleteInterfaceDiagramElementUndoRedoTest() {
+		addElement(Interface.class, conceptFuncElectrical, elementConfiguration);
+		dragTreeItemToDiagramEditor(elementConfiguration, diagramEditor);
+		Assert.assertTrue(isEditPartPresentInDiagramEditor(diagramEditor, "ElementConfiguration"));
+		
+		String elementConfigurationName = elementConfiguration.getText();
+		
+		deleteEditPartInDiagramEditor(diagramEditor, "ElementConfiguration");
+		Assert.assertFalse(isEditPartPresentInDiagramEditor(diagramEditor, "ElementConfiguration"));
+		Assert.assertFalse(isTreeItemPresentInTreeView(elementConfiguration));
+		
+		diagramEditor.setFocus();
+		
+		undoCommand();
+		Assert.assertTrue(isEditPartPresentInDiagramEditor(diagramEditor, "ElementConfiguration"));
+		
+		SWTBotTreeItem elementConfigurationNode = configurationTree.getNode(elementConfigurationName);
+		Assert.assertNotNull(elementConfigurationNode);
+		
+		diagramEditor.setFocus();
+
+		redoCommand();
+		Assert.assertFalse(isEditPartPresentInDiagramEditor(diagramEditor, "ElementConfiguration"));
+		Assert.assertFalse(isTreeItemPresentInTreeView(elementConfigurationNode));
+	}
+	
+	@Test
+	public void connectInterfaceEndsTest() {		
+		addElement(Interface.class, conceptFuncElectrical, elementConfiguration);
+		
+		dragTreeItemToDiagramEditor(elementConfiguration, diagramEditor, DRAG1_COORDINATES_X, DRAG1_COORDINATES_Y);
+		SWTBotGefEditPart swtBotGefEditPart1 = diagramEditor.selectedEditParts().get(0);		
+
+		dragTreeItemToDiagramEditor(elementConfiguration, diagramEditor, DRAG2_COORDINATES_X, DRAG2_COORDINATES_Y);
+		SWTBotGefEditPart swtBotGefEditPart2 = diagramEditor.selectedEditParts().get(0);		
+		
+		diagramEditor.activateTool("InterfaceEnd");
+		diagramEditor.click(swtBotGefEditPart1);
+		
+		diagramEditor.activateTool("InterfaceEnd");
+		diagramEditor.click(swtBotGefEditPart2);
+		
+		diagramEditor.activateTool("Interface");
+		diagramEditor.click(swtBotGefEditPart1.children().get(0));
+		diagramEditor.click(swtBotGefEditPart2.children().get(0));
+		bot.button("OK").click();
+		
+		List<SWTBotGefConnectionEditPart> sourceConnections = swtBotGefEditPart1.children().get(0).children().get(0).sourceConnections();
```

Comment:
Here as well, maybe we can get some clearer names.
I think here the sourceConnections are the interfaces?